azavea/temperate
# Bootstrap Area Indicators

## Overview

This PR migrates the Scala project from azavea/climate-backend to a subproject folder here in the Temperate repository. It also sets up a basic SBT build and a container configured to run the assembly jar as part of the docker-compose stack. Like other projects, its expected that developers will compile/test/debug the SBT project natively on their machines -- VM/Docker filesystem performance remains subpar.

This PR only adds the the new area-indicators container to the `update` and `server` scripts to serve development environment interests. Integration of the area-indicators container into the deployment scripts can be done at a later date.

~~I'm going to continue working forward from this branch. I'd like to also address #1229 before the end of the sprint and use that task to modernize the SBT build.sbt definition. But that's not worth doing until we actually have the libraries we're going to use, namely http4s and circe rather than akka/spray so I didn't address it here.~~ **Edit:**: Complete. The last few commits on this branch migrate the project to a modern http4s setup and swap the now commented out (for reference) Akka Router/Server with one from http4s. With an aws profile set, it queries the S3 layer but fails with an out of memory error. I'm not sure how this is different from the old prototype, need to talk to James to determine whether to address this now or in a separate issue. Likely a separate issue, since getting this merged is the first step towards parallelizing tasks, and there are a number of them that don't require a working indicators endpoint.

### Demo

![Screen Shot 2019-07-02 at 7 26 22 PM](https://user-images.githubusercontent.com/1818302/60553010-50574180-9cff-11e9-9db4-b362ceb59028.png)

### Notes

The `backend` project was renamed to `api` to follow conventions of other projects. We'll add other subprojects as they spin off from the concern of "handle web requests and return responses". The api package name was also updated from `com.azavea.climate.backend` to `io.temperate.api`.

## Testing Instructions

Ensure that Java 1.8+ is installed on your machine. Then, run `./scripts/update` and `./scripts/server` which should start the area indicators http4s server at http://localhost:8108/indicators/

The following should also work:
```bash
pushd src/area-indicators
AWS_PROFILE=planit ./sbt "api/run"
```


Closes #1228 
Closes #1229
